.github/workflows/pr-preview.yml

```
@@ -0,0 +1,51 @@
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+      - name: Validate schema and run generator tests
+        if: github.event.action != 'closed'
+        run: |
+          node web/js/schema.test.mjs
+          node web/js/state.test.mjs
+          node web/js/generate.test.mjs
+      - name: Point GitHub links at this PR's branch
+        if: github.event.action != 'closed'
+        run: bash .github/scripts/stamp-ref.sh "${{ github.head_ref }}"
```

actionlint (1.7.12)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)